### DISCUSSION TITLE
How does TransformManager work

### DISCUSSION THREAD

**Discussion by yicong123456** [google/filament#7376] [Q&A]
I used filament on Android and found that TransformManager is really easy to use,  really curious about how it implemented. And wanna know whether the transformmanager do transform in CPU or GPU, and if it do the transform in CPU,what is the advantage and disadvantage of using TransformManager and passing vertex data into vertex shader and do rotation or move using matrix multiplication. Thanks so much!

**Comment by romainguy** [comment]
All the transforms are applied to the vertices on the GPU.

**Reply by ywang218** [reply]
Got it, thanks for the explanation!
